tests/test_factory_fixtures.py

```diff
@@ -190,3 +195,22 @@ def test_lazy_fixture_post_generation(author):
     """Test that post-generation values are replaced with lazy fixtures."""
     # assert author.user.username == "lazyfixture"
     assert author.user.password == "asdasd"
+
+
+def test_register_class_decorator_with_kwargs_only(harry_potter_author):
+    """Ensure ``register`` decorator called with kwargs only works normally."""
+    assert harry_potter_author.name == "J.K. Rowling"
+    assert harry_potter_author.user.username == "jk_rowling"
+
+
+register(_name="the_chronicles_of_narnia_author", name="C.S. Lewis")(
+    AuthorFactory,
+    register_user="cs_lewis",
+    register_user__password="Aslan1",
+)
```

Comment:
not sure we want to support an invocation like this 🤣. I'd just not bother testing this.

Reply:
I agree (and I'd really like your idea of using a dedicated `dict` parameter instead of `kwargs`), but this is just used for testing purposes.